src/app/decorators/emit-on-destroy.ts

```diff
@@ -4,15 +4,33 @@ import {interposeAfterFunction, interposeBeforeFunction} from './helpers/decorat
 /**
  * EmitOnDestroy creates a subject which emits once and immediately completes after ngOnDestroy was called.
  * This can be used to easily implement the takeUntil() pattern for subscriptions which have to be canceled on a component's destroy hook.
+ * 
+ * For angular based frameworks that have their own lifecycle-hooks like e.g. ionic, hooks can be specified via configuration.
+ * Example Usage:
+ * @EmitOnDestroy({
+ *  initialize: 'ionViewDidEnter',
+ *  terminate: 'ionViewDidLeave'
+ * })
+ * 
  */
 
-export function EmitOnDestroy() {
+export interface IEmitOnDestroyConfig {
```

Comment:
- i am using microsoft's internal typescript style guide. please remove the leading I from the interface
(https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines)
- please move the interface to an own file within a new directory "interfaces"